src/sentry/runner/commands/configoptions.py

```diff
@@ -16,8 +18,60 @@
 UNSET_MSG = "[UNSET] Option %s unset."
 
 
+class OptionsPresenterController:
+    def __init__(self) -> None:
+        pass
+
+    def check_slack_webhook_config(self):
+        return True
+
+    consolepresenter = ConsolePresenter()
+    slackpresenter = SlackPresenter()
+    drifted_options = []
+    channel_updated_options = []
+    updated_options = []
+    set_options = []
+    unset_options = []
+
+    def update(self, key: str, db_value: Any, value: Any):
+        self.updated_options.append((key, db_value, value))
+
+    def channel_update(self, key: str):
+        self.channel_updated_options.append(key)
+
+    def drift(self, key: str):
+        self.drifted_options.append(key)
+
+    def set(self, key: str, value: Any):
+        self.set_options.append((key, value))
+
+    def unset(self, key: str):
+        self.unset_options.append(key)
+
+    def write(self):
+        if self.check_slack_webhook_config:
+            self.slackpresenter.write()
+        self.consolepresenter.write(
+            self.drifted_options,
+            self.channel_updated_options,
+            self.updated_options,
+            self.set_options,
+            self.unset_options,
+        )
```

Comment:
This is changing the semantics with respect to what happens today. 
Today, if we encounter an error, we keep writing log lines up to that point (which could have updated options) and then crash when we run into the error.
In this case (since all writes happen at the end) if there is an error at any point, the preceding output is lost. 

Is that intentional? If yes I think we should reconsider. The preceding output was useful. Though we need to be able to handle both dumping the output that precedes the error on the CLI and dumping writing the output properly at the end for slack.

Reply:
Yeah, I realized that while I was building this that I was changing the tool into an all or nothing kind of deal, there are trade offs with each.